pkg/scheduler/internal/cache/cache.go

```diff
@@ -715,8 +715,24 @@ func (cache *schedulerCache) GetNodeInfo(nodeName string) (*v1.Node, error) {
 
 	n, ok := cache.nodes[nodeName]
 	if !ok {
-		return nil, fmt.Errorf("error retrieving node '%v' from cache", nodeName)
+		return nil, fmt.Errorf("node %q not found in cache", nodeName)
 	}
 
 	return n.info.Node(), nil
 }
+
+// ListNodes returns the cached list of nodes.
+func (cache *schedulerCache) ListNodes() ([]*v1.Node, error) {
```

Comment:
It seems `error` is unnecessary.

Reply:
It is necessary for some implementations. But also this should be discussed in the parent PR.